### PR TITLE
Reusable cluster deployer pipeline

### PR DESCRIPTION
**Pull request by blairboy362**
## What

Adds a reusable deployer pipeline and corresponding terraform template that can be used to `fly set-pipeline` and spin up a cluster

This is a step towards us managing clusters soley from concourse.

## How to test

* Check out this branch
* Push a pipeline that will provision a cluster in sandbox account called $MY_CLUSTER to the concourse with fly like:
   ```
    fly -t cd-gsp set-pipeline -p sandbox --config pipelines/deployer.yaml --var account-name=sandbox --var account-role-arn=arn:aws:iam::011571571136:role/deployer --var cluster-name=$MY_CLUSTER --yaml-var trusted-developer-keys="$(yq . ./users/*.yaml | jq -s '[ .[].pub ]')" --var splunk_hec_token="NOTATOKEN" --var github-client-secret=NOTASECRET --var github-client-id=NOTID --var gsp-version=eks-firebreak --check-creds
    ```
* Unpause / run the pipeline
* Wait till it's green
* Request your kubeconfig like:
   ```
   aws-vault exec admin@sandbox -- aws eks update-kubeconfig --name $MY_CLUSTER --kubeconfig ./kubeconfig
    ```
* Set your kubeconfig:
    ```
    export KUBECONFIG=$(pwd)/kubeconfig
    ```
* play with your gsp cluster:
    ```
    aws-vault exec admin@sandbox -- kubectl get po --all-namepaces
    ```

##### Note

* should remove references to `eks-firebreak` branch before merging
* @chrisfarms re-wrote this PR text... after continuing the work @blairboy362 started on a pipeline for the EKS branch of GSP ... apologies for editing words into your face

### CODE REVIEW

File: terraform/accounts/run-sandbox/persistent/drb-eks-test.run-sandbox.aws.ext.govsandbox.uk/resources.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module "gsp-network" {
+  source       = "git::https://github.com/alphagov/gsp-terraform-ignition//modules/gsp-network?ref=eks-firebreak"
+  cluster_name = "samcrang"
```
Comment (chrisfarms) [inline terraform/accounts/run-sandbox/persistent/drb-eks-test.run-sandbox.aws.ext.govsandbox.uk/resources.tf]: sam's

Comment (samcrang) [reply terraform/accounts/run-sandbox/persistent/drb-eks-test.run-sandbox.aws.ext.govsandbox.uk/resources.tf]: hello

File: pipelines/terraform/cluster/main.tf [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  cluster_name = "${var.cluster_name}"
+  dns_zone     = "${var.dns_zone}"
+
+  admin_role_arns = [
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/admin",
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/deployer",
+  ]
+
+  sre_user_arns = [
+    "arn:aws:iam::622626885786:user/daniel.blair@digital.cabinet-office.gov.uk"
```
Comment (blairboy362) [inline pipelines/terraform/cluster/main.tf]: Just me?

File: pipelines/terraform/cluster/main.tf [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  dns_zone     = "${var.dns_zone}"
+}
+
+module "gsp-network" {
+  source       = "git::https://github.com/alphagov/gsp-terraform-ignition//modules/gsp-network"
+  cluster_name = "${var.cluster_name}"
+}
+
+module "gsp-cluster" {
+  source       = "git::https://github.com/alphagov/gsp-terraform-ignition//modules/gsp-cluster"
```
Comment (blairboy362) [inline pipelines/terraform/cluster/main.tf]: Is it worth adding a `module_ref` variable to make it easier to create a cluster testing a branch of `gsp-terraform-ignition`?

File: pipelines/terraform/common/admins.tf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      identifiers = "${concat(var.additional_admins, list(
+        "arn:aws:iam::622626885786:user/sam.crang@digital.cabinet-office.gov.uk",
+        "arn:aws:iam::622626885786:user/daniel.blair@digital.cabinet-office.gov.uk",
+        "arn:aws:iam::622626885786:user/chris.farmiloe@digital.cabinet-office.gov.uk",
+        "arn:aws:iam::622626885786:user/rafal.proszowski@digital.cabinet-office.gov.uk",
+        "arn:aws:iam::622626885786:user/david.pye@digital.cabinet-office.gov.uk",
+        "arn:aws:iam::622626885786:user/stephen.ford@digital.cabinet-office.gov.uk",
+        "arn:aws:iam::622626885786:user/david.povey@digital.cabinet-office.gov.uk",
+        "arn:aws:iam::622626885786:user/david.mcdonald@digital.cabinet-office.gov.uk",
+        "arn:aws:iam::622626885786:user/toby.lornewelch-richards@digital.cabinet-office.gov.uk",
```
Comment (blairboy362) [inline pipelines/terraform/common/admins.tf]: Toby may not need to be here any more?

Comment (paroxp) [reply pipelines/terraform/common/admins.tf]: Neither does anyone else to be fair. I'd say let's take a risk and limit the list to two/three users in general, promote the SRE role instead.

File: pipelines/terraform/cluster/main.tf (hunk 3 above)
Comment (chrisfarms) [reply pipelines/terraform/cluster/main.tf]: Good idea ...

File: pipelines/terraform/common/admins.tf (hunk 4 above)
Comment (chrisfarms) [reply pipelines/terraform/common/admins.tf]: this file can actually just disapear for now

File: pipelines/terraform/cluster/main.tf (hunk 3 above)
Comment (chrisfarms) [reply pipelines/terraform/cluster/main.tf]: added, but not tested

**Pull request merged**